### DISCUSSION TITLE
issues trying to push to Github from Android Studio

### DISCUSSION THREAD

**Discussion by AndroidSDK4Ever** [community/community#136834] [Pull Requests] [answered]
Hi veryone,

I'm having some trouble pushing my project to GitHub directly from Android Studio, and I'm hoping someone can help me figure out what I'm doing wrong.

I Initialized Git in my project, committed my changes using VCS > Commit..., added a commit message, and everything seemed to commit properly.

I added my GitHub repository as a remote using VCS > Git > Remotes..., entered the correct URL and tried to push.
I attempted to push my changes by going to VCS > Git > Push.... This is where I’m encountering issues.

When I try to push, I’m getting an error message related to authentication or permissions (sometimes it just says "Push failed" without much detail).
I’ve checked that I’m authenticated to GitHub in Android Studio (I’m using HTTPS, not SSH).
I’ve also made sure that I have the necessary permissions on the GitHub repository.

**Comment by agustrodmar** [comment]
Hi @AndroidSDK4Ever 

I'm assuming you have searched for this in the forum. Sometimes, push can be quite confusing for beginners while using Android Studio.

I would need more details, however, I'll recommend you to use the integrated terminal in Android Studio, commit and push feel much better and easier this way. Trust me. It may be a little bit annoying at first, but after a few times you'll appreaciate this. 

Open the console below in Android studio and:

`git add .
git commit -m "Your commit message here"
git push origin main`

IMPORTANT: I'm assuming you have not pushed anything yet. 

If this works, it confirms that the issue is likely with how Android Studio is interacting with your Git account. If this doesn't work, check your authentication settings.

**Comment by AndroidSDK4Ever** [comment]
Thanks for your fast reply and the suggestions. I tried using the terminal in Android Stuio, however when I run git push, the terminal prompts me for my Github username and password, despite I introduced it properly it keeps failing with an authentication error. I’m sure the password is correct because I can log in to GitHub via the website without any issues.

Is there something I’m missing here? Why isn’t my password being accepted, and how can I resolve this?

**Comment by agustrodmar** [comment, marked as answer]
Ok! that sounds like a really common error. The issue is likely due to GitHub’s recent changes to their authentication process. GitHub no longer supports using your account password for Git operations (like pushing code) when using HTTPS. Instead, you need to use a Personal Access Token (PAT).

Go to:
https://github.com/settings/tokens

Check the boxes for the permissions you need, then select Generate Token and save it for the future and keep it safe.  

If you don't want to introduce the token again and again you could cache it:
`git config --global credential.helper store`
After running this, next time you enter the token it will be stored. 
Give this a try, and you should be able to push your changes without any issues! Let me know if you run into any more problems.

**Reply by AndroidSDK4Ever** [reply]
It did work. Thanks a lot!
